lib/python/vdm/server/HTTPListener.py

```diff
@@ -499,6 +499,127 @@ def get_volt_jar_dir():
     return os.path.realpath(os.path.join(Global.MODULE_PATH, '../../../..', 'voltdb'))
 
 
+def get_port(ip_with_port):
+    if ip_with_port != "":
+        if ":" not in ip_with_port:
+            return ip_with_port
+        else:
+            arr = ip_with_port.split(":")
+            return arr[1]
+
+
+def check_port_valid(port_option, servers):
+    if port_option == "http-listener":
+        default_port = "8080"
+    if port_option == "admin-listener":
+        default_port = "21211"
+    if port_option == "zookeeper-listener":
+        default_port = "7181"
+    if port_option == "replication-listener":
+        default_port = "5555"
+    if port_option == "client-listener":
+        default_port = "21212"
+    if port_option == "internal-listener":
+        default_port = "3021"
+    if port_option not in request.json:
+        port = default_port
+        server_port = get_port(servers[port_option])
```

Comment:
These 3 lines (526-528) should be taken out of this if-else and put after the if-else. This is not dependent on whether port_option is specified in the request son or not.